### ISSUE TITLE
[StyleGAN] A Style-Based Generator Architecture for Generative Adversarial Networks (Mar 2019 CVPR)

### ISSUE THREAD

**Issue by andrewjong** [andrewjong/Deep-Learning-Paper-Surveys#17]
## 0. Article Information and Links

- Paper's project website: https://github.com/NVlabs/stylegan
- Release date: YYYY/MM/DD
- Number of citations (as of 2020/MM/DD): 

### Architecture

![image](https://user-images.githubusercontent.com/8121216/86195730-6bbd7e00-bb06-11ea-835f-ae4d285feaee.png)

### Learn to transform latent code Z to intermediate latent space W
Argues that Z will often be entangled, learning to transform to W will untangle. Notes that deeper layers generally result in less entangled directions.

### New style norm layer: Adaptive Instance Normalization (AdaIN)
Enables styles.

### Perceptual Path Length
Longer paths imply more curvature and irregularity in the space, meaning greater entanglement. Shorter paths means the space is more regular. In the image below, the authors argue latent code Z is more entangled, whereas intermediate latent space W can learn to untangle.
![image](https://user-images.githubusercontent.com/8121216/86192353-4462b300-bafe-11ea-828f-2d185672cccf.png)

### Linear Separability
Train a _linear_ SVM to find hyperplane separating the binary attributes.

CelebA has 40 binary attributes, e.g. male/female, glasses/no-glasses.

Train separate MLPs (same architecture as discriminator) to classify each binary attribute. Then train linear SVMs to do the same. Take the conditional cross entropy between the two predictors.

![image](https://user-images.githubusercontent.com/8121216/86195932-e090b800-bb06-11ea-900a-075f5ba0033d.png)

To measure the separability of one attribute, we generate 200,000 images with z ∼ P (z) and classify them using the auxiliary classification network. We then sort the samples according to classifier confidence and remove the least confident half, yielding 100,000 labeled latent-space vectors.

Compute the conditional entropy H(Y|X) where X are the classes predicted by the SVM and Y are the classes determined by the pre-trained classifier. This tells how much additional information is required to determine the true class of a sample, given that we know on which side of the hyperplane it lies. A low value suggests consistent latent space directions for the corresponding factor(s) of variation.

## 6. Things to discuss? (e.g. weaknesses, potential for future work, relation to other work)

The paper proposes _two_ new measurements of disentanglement:
1) Perceptual Path Length

2) Linear Separability. 
Both are extremely useful because disentanglement is important for higher user control.

How would we measure linear separability for data without binary attributes, e.g. unlabeled data?
For garments, maybe sleeve length?
